tls-codec/src/primitives.rs

```diff
@@ -0,0 +1,73 @@
+use super::{Cursor, Deserialize, Error, Serialize};
+
+use std::convert::TryInto;
+
+impl Serialize for u8 {
+    fn tls_serialize(&self, buffer: &mut Vec<u8>) -> Result<(), Error> {
+        buffer.push(*self);
+        Ok(())
+    }
+}
+
+impl Serialize for u16 {
+    fn tls_serialize(&self, buffer: &mut Vec<u8>) -> Result<(), Error> {
+        buffer.extend_from_slice(&self.to_be_bytes());
+        Ok(())
+    }
+}
+
+impl Serialize for u32 {
+    fn tls_serialize(&self, buffer: &mut Vec<u8>) -> Result<(), Error> {
+        buffer.extend_from_slice(&self.to_be_bytes());
+        Ok(())
+    }
+}
+
+impl Serialize for u64 {
+    fn tls_serialize(&self, buffer: &mut Vec<u8>) -> Result<(), Error> {
+        buffer.extend_from_slice(&self.to_be_bytes());
+        Ok(())
+    }
+}
+
+impl Deserialize for u8 {
+    fn tls_deserialize(cursor: &Cursor) -> Result<Self, Error> {
+        match cursor.read(1) {
```

Comment:
Wouldn't it be more succinct to use `map()` here? (also applies to the following functions)

Reply:
Here it would be indeed. The others use `try_into`, which doesn't work in the closure.